lib/timeline.js

```diff
@@ -0,0 +1,107 @@
+class Timeline {
+  constructor(client) {
+    this.client = client
+  }
+
+  createEventType(applicationId, userId, data, cb) {
+    return this.client._request(
+      {
+        method: 'POST',
+        path: `/integrations/v1/${applicationId}/timeline/event-types?userId=${userId}`,
+        body: data,
+      },
+      cb,
+    )
+  }
+
+  updateEventType(eventId, applicationId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createEventTypeProperty(eventId, applicationId, userId, data, cb) {
+    const parameters = {
+      method: 'POST',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties?userId=${userId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  updateEventTypeProperty(eventId, applicationId, propertyId, data, cb) {
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties/${propertyId}`,
```

Comment:
Looks like this endpoint is not consistent with other hubspot endpoints. It seems to be a PUT to `/integrations/v1/${applicationId}/timeline/event-types/${eventId}/properties` and the id needs to be in the body of the request.

You could do something like:

```js
updateEventTypeProperty(eventId, applicationId, propertyId, data, cb) {
  data['id'] = data['id'] || propertyId
  const parameters = {
    method: 'PUT',
    path: `/integrations/v1/${applicationId}/timeline/event-types/${eventTypeId}/properties`,
    body: data,
  }

  return this.client._request(parameters, cb)
}
```

https://developers.hubspot.com/docs/methods/timeline/udpate-timeline-event-type-property